src/libsyntax/test.rs

```diff
@@ -142,9 +140,18 @@ impl<'a> fold::Folder for TestHarnessGenerator<'a> {
                     self.tests.push(i.ident);
                     // debug!("have {} test/bench functions",
                     //        cx.testfns.len());
+
+                    // Make all tests public so we can call them from outside
+                    // the module (note that the tests are re-exported and must
+                    // be made public themselves to avoid privacy errors).
+                    let mut result = (*i).clone();
```

Comment:
You're never supposed to clone AST nodes if you own them (can we make specific impls of `Clone` warn/error?), here's what you can do instead:
```rust
i.map(|mut i| { i.vis = ast::Public; i })
```